File: daym-driver/app/src/main/java/com/gtpautomation/driver/api_services/VolleyErrorResolver.java
package com.gtpautomation.driver.api_services;

import com.android.volley.AuthFailureError;
import com.android.volley.NoConnectionError;
import com.android.volley.TimeoutError;
import com.android.volley.VolleyError;

import org.json.JSONException;
import org.json.JSONObject;

public class VolleyErrorResolver {
    public static CustomError getErrorMessage(VolleyError error) {
        if (error instanceof TimeoutError || error instanceof NoConnectionError) {
            return new CustomError("Unable to connect to server", 0);
        } else if (error instanceof AuthFailureError) {
            return new CustomError("Something went wrong please login to continue", 401);
        } else if (error.networkResponse != null) {

            switch (error.networkResponse.statusCode) {
                case 409:
                case 400:
                    try {
                        String badUrlResponse = new JSONObject(new String(error.networkResponse.data)).getString("message");
                        return new CustomError(badUrlResponse, error.networkResponse.statusCode);
                    } catch (JSONException ignored) {
                        return new CustomError("Unknown Error Occurred", 1);
                    }
                case 401:
                    return new CustomError("Something went wrong please login to continue", 401);
                case 404:
                    return new CustomError("Unable to connect to server", error.networkResponse.statusCode);
                case 413:
                    return new CustomError("File size too large", 413);
                case 500:
                    try {
                        String five_hundred_error = new JSONObject(new String(error.networkResponse.data)).getString("message");
                        return new CustomError(five_hundred_error, error.networkResponse.statusCode);
                    } catch (JSONException ignored) {
                    }
                default:
                    return new CustomError("Some error occurred, please try again later", error.networkResponse.statusCode);
            }
        } else
            return new CustomError("Unknown Error Occurred", 1);
    }

    public static class CustomError {
        private String message;
        private int statusCode;

        public CustomError() {
        }

        public CustomError(String message, int statusCode) {
            this.message = message;
            this.statusCode = statusCode;
        }

        public String getMessage() {
            return message;
        }

        public void setMessage(String message) {
            this.message = message;
        }

        public int getStatusCode() {
            return statusCode;
        }

        public void setStatusCode(int statusCode) {
            this.statusCode = statusCode;
        }
    }
}

File: daym-security-guard/app/src/main/java/com/gtpautomation/securityguard/utils/apiCall/VolleyErrorResolver.kt
package com.gtpautomation.securityguard.utils.apiCall

import android.util.Log
import com.android.volley.AuthFailureError
import com.android.volley.NoConnectionError
import com.android.volley.TimeoutError
import com.android.volley.VolleyError
import com.gtpautomation.securityguard.utils.Constants
import org.json.JSONException
import org.json.JSONObject

class VolleyErrorResolver {
    companion object{
        fun getErrorMessage(error: VolleyError): CustomError? {
            return if (error is TimeoutError || error is NoConnectionError) {
                CustomError("Unable to connect to server", Constants.DEFAULT_ERROR_STATUS_CODE)
            } else if (error is AuthFailureError) {
                CustomError(
                    "Something went wrong please login to continue",
                    Constants.AUTH_ERROR_STATUS_CODE
                )
            } else if (error.networkResponse != null) {
                Log.e("VolleyErrorResolver", "getErrorMessage: "+error.networkResponse.data.contentToString(), )
                when (error.networkResponse.statusCode) {
                    409, 400 -> {
                        return try {
                            val badUrlResponse =
                                JSONObject(String(error.networkResponse.data)).getString("message")
                            CustomError(badUrlResponse, error.networkResponse.statusCode)
                        } catch (ignored: JSONException) {
                            CustomError("Unknown Error Occurred", 1)
                        }
                    }
                    401 -> CustomError(
                        "Something went wrong please login to continue",
                        Constants.AUTH_ERROR_STATUS_CODE
                    )
                    404 -> CustomError(
                        "Unable to connect to server",
                        error.networkResponse.statusCode
                    )
                    413 -> CustomError("File size too large", 413)
                    500 -> {
                        try {
                            val five_hundred_error =
                                JSONObject(String(error.networkResponse.data)).getString("message")
                            //Log.e("error_500",five_hundred_error);
                            return CustomError(five_hundred_error, error.networkResponse.statusCode)
                        } catch (ignored: JSONException) {
                        }
                        CustomError(
                            "Some error occurred, please try again later",
                            error.networkResponse.statusCode
                        )
                    }
                    else -> CustomError(
                        "Some error occurred, please try again later",
                        error.networkResponse.statusCode
                    )
                }
            } else CustomError("Unknown Error Occurred", 1)
        }

        class CustomError {
            var message: String? = null
            var statusCode = 0

            constructor() {}
            constructor(message: String?, statusCode: Int) {
                this.message = message
                this.statusCode = statusCode
            }
        }
    }
}